src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ReadMetadata.java

```diff
@@ -254,49 +299,72 @@ public ReadMetadata read( final Kryo kryo, final Input input, final Class<ReadMe
     @DefaultSerializer(PartitionStatistics.Serializer.class)
     public static final class PartitionStatistics {
         private final static int MAX_TRACKED_FRAGMENT_LENGTH = 10000;
-        private final Map<String, long[]> readGroupToFragmentSizeCountMap;
+        private final Map<String, long[]> libraryToFragmentSizeCountMap;
         private final long nReads;
         private final long nBases;
-
-        public PartitionStatistics( final Iterator<GATKRead> readItr, final int nReadGroups ) {
-            readGroupToFragmentSizeCountMap = new HashMap<>(SVUtils.hashMapCapacity(nReadGroups));
-            long reads = 0L;
-            long bases = 0L;
-            while ( readItr.hasNext() ) {
-                final GATKRead read = readItr.next();
-                reads += 1L;
-                bases += read.getLength();
-                if ( read.isFirstOfPair() && !read.isSecondaryAlignment() && !read.isSupplementaryAlignment() &&
-                        !read.isUnmapped() && !read.mateIsUnmapped() &&
-                        Objects.equals(read.getContig(), read.getMateContig()) ) {
-                    int tLen = Math.abs(read.getFragmentLength());
-                    if ( tLen > MAX_TRACKED_FRAGMENT_LENGTH ) tLen = MAX_TRACKED_FRAGMENT_LENGTH;
-                    final String readGroup = read.getReadGroup();
-                    final long[] counts = readGroupToFragmentSizeCountMap.computeIfAbsent(readGroup,
-                                                              k -> new long[MAX_TRACKED_FRAGMENT_LENGTH+1]);
-                    counts[tLen] += 1;
+        private final String firstContig;
+        private final int firstLocation;
+        private final String lastContig;
+        private final int lastLocation;
+
+        public PartitionStatistics( final Iterator<GATKRead> readItr, final Map<String, String> readGroupToLibraryMap ) {
+            if ( !readItr.hasNext() ) {
+                libraryToFragmentSizeCountMap = Collections.emptyMap();
+                nReads = 0L;
+                nBases = 0L;
+                firstContig = null;
+                firstLocation = 0;
+                lastContig = null;
+                lastLocation = 0;
+            } else {
+                long reads = 0L;
+                long bases = 0L;
+                GATKRead read = readItr.next();
+                firstContig = read.getContig();
+                firstLocation = read.getStart();
+                libraryToFragmentSizeCountMap = new HashMap<>(SVUtils.hashMapCapacity(readGroupToLibraryMap.size()));
+                while ( true ) {
```

Comment:
Because I've already grabbed the first read from the iterator (to get the firstContig and firstLocation data), but I still have to account for its reads, bases, and tLen.  So I have to enter the loop once, even when there isn't an additional read to process.  I could express it as a do/while, but I'm not sure it would be any clearer:
```
    GATKRead read = mappedReadItr.next();
    firstContig = read.getContig();
    firstLocation = read.getStart();
    do {
        a bunch of stuff
    } while ( (read = (readItr.hasNext() ? readItr.next() : null)) != null );
```
Do you prefer the do/while form?